src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Platform/LoggerBase.cs

```diff
@@ -50,7 +50,15 @@ internal abstract class LoggerBase
                 DefaultLog(logLevel, formattedMessage);
             }
 
-            LoggerCallbackHandler.ExecuteCallback(logLevel, formattedMessage, containsPii);
+            if (LoggerCallbackHandler.LogCallback != null)
+            {
+                LoggerCallbackHandler.ExecuteCallback(logLevel, formattedMessage, containsPii);
+            }
+            else if (!containsPii)
+            {
+                // execute obsolete IAdalLogCallback only if LogCallback is not set and message does not contain Pii
+                LoggerCallbackHandler.ExecuteCallback(logLevel, formattedMessage);
```

Comment:
code is confusing to read. We should either use different name for methods of rename to old function to contain the word obsolete.